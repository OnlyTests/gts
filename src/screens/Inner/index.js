import {Container} from 'native-base';
import React from 'react';
import {View, Text} from 'react-native';
import {Actions} from 'react-native-router-flux';
import {PersonSuporte} from '../../assets/svgs/illustration';
import ButtonCustom from '../../components/buttons/buttonCustom';
import colors from '../../styles/colors';
import st from './styles';
function Inner() {
  
  function goToLogin() {
    Actions.login();
  }

  function goToRegister() {
    Actions.register();
  }
  return (
    <Container style={st.root}>
      <View style={st.align}>
        <Text style={st.titleTxt}>GTS Cursos & desafios</Text>
        <View style={st.illustration}>
          <PersonSuporte />
        </View>
      </View>
      <View style={st.flex} />
      <ButtonCustom onPress={goToLogin} label="Login" />
      <ButtonCustom
        background="transparent"
        labelColor={colors.primary}
        borderWidth={2}
        onPress={goToRegister}
        label="Cadastro"
      />
    </Container>
  );
}

export default Inner;
