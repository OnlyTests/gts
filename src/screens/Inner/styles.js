import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';

export default StyleSheet.create({
  root: {
    backgroundColor: colors.base,
    paddingHorizontal: 16,
  },
  flex: {flex: 1},

  align: {
    alignItems: 'center',
  },
  titleTxt: {
    fontSize: 24,
    paddingTop: 10,
    fontWeight: 'bold',
    color: colors.primary,
  },
  illustration: {
    width: '70%',
    height: '70%',
  },
});
