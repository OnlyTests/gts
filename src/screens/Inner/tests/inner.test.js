import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Inner from '../index';

describe('Inner, check if render success', () => {
  afterEach(jest.clearAllMocks);
  jest.useFakeTimers();

  test('render', async () => {
    const {toJSON, unmount} = render(<Inner />);
    expect(toJSON()).toMatchSnapshot();
    await unmount();
  });

  test('check if login buttons are works', async () => {
    const {unmount, getByText} = render(<Inner />);
    //check if have texts
    expect(getByText('GTS Cursos & desafios')).toBeDefined();
    expect(getByText('Cadastro')).toBeDefined();
    expect(getByText('Login')).toBeDefined();
    fireEvent.press(getByText('Login'));
    await unmount();
  });
  test('check if register buttons are works', async () => {
    const {unmount, getByText} = render(<Inner />);
    fireEvent.press(getByText('Cadastro'));
    await unmount();
  });
});
