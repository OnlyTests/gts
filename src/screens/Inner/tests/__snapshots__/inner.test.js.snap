// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inner, check if render success render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#F5F6F7",
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2f5199",
          "fontSize": 24,
          "fontWeight": "bold",
          "paddingTop": 10,
        }
      }
    >
      GTS Cursos & desafios
    </Text>
    <View
      style={
        Object {
          "height": "70%",
          "width": "70%",
        }
      }
    >
      <svg
        data-file-name="SvgPersonSuporte"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2f5199",
        "borderColor": "#dadada",
        "borderRadius": 40,
        "borderWidth": 0.3,
        "flexDirection": "row",
        "height": 52,
        "justifyContent": "center",
        "marginHorizontal": 0,
        "marginLeft": 0,
        "marginVertical": 5,
        "opacity": null,
        "paddingHorizontal": 24,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": undefined,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#dadada",
        "borderRadius": 40,
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 52,
        "justifyContent": "center",
        "marginHorizontal": 0,
        "marginLeft": 0,
        "marginVertical": 5,
        "opacity": null,
        "paddingHorizontal": 24,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2f5199",
            "fontSize": undefined,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        Cadastro
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
